examples/meta/src/kernel/weighted_degree_string.sg.in

```diff
@@ -0,0 +1,25 @@
+File traindat = read_csv("@SHOGUN_DATA@/fm_train_dna.dat")
+File testdat = read_csv("@SHOGUN_DATA@/fm_test_dna.dat")
+
+#![create_features]
+Features feats_train = create_char_string_features(traindat, enum EAlphabet.DNA)
+Features feats_test = create_char_string_features(testdat, enum EAlphabet.DNA)
+#![create_features]
+
+#![create_kernel]
+int degree = 3
+WeightedDegreeStringKernel kernel(degree)
```

Comment:
when i write `WeightedDegreeStringKernel kernel(feats_train, feats_train, degree)`, i got some errors.

>  File "kernel_weighted_degree_position_string copy.py", line 14, in <module>
    kernel = sg.WeightedDegreeStringKernel(feats_train, feats_train,degree)
TypeError: Wrong number or type of arguments for overloaded function 'new_WeightedDegreeStringKernel'.
  Possible C/C++ prototypes are:
    shogun::WeightedDegreeStringKernel::WeightedDegreeStringKernel()
    shogun::WeightedDegreeStringKernel::WeightedDegreeStringKernel(int32_t,shogun::EWDKernType)
    shogun::WeightedDegreeStringKernel::WeightedDegreeStringKernel(int32_t)
    shogun::WeightedDegreeStringKernel::WeightedDegreeStringKernel(shogun::SGVector< float64_t >)
    shogun::WeightedDegreeStringKernel::WeightedDegreeStringKernel(std::shared_ptr< shogun::StringFeatures< char > > const &,std::shared_ptr< shogun::StringFeatures< char > > const &,int32_t)

but ` shogun::WeightedDegreeStringKernel::WeightedDegreeStringKernel(std::shared_ptr< shogun::StringFeatures< char > > const &,std::shared_ptr< shogun::StringFeatures< char > > const &,int32_t)` is the needed constructor, I can't figure out why it didn't work?

Reply:
the issue is that you pass `feats_train` which is of type `std::shared_ptr<Features>`, but the constructor requires the derived type `std::shared_ptr< shogun::StringFeatures<char>>`. But passing the features to `kernel.init` works fine right?